UnityProject/Assets/Scripts/Transform/CustomNetTransform.Move.cs

```diff
@@ -0,0 +1,708 @@
+	{
+		if (!float.IsNaN(speed) && speed <= 0)
+		{
+			return false;
+		}
+
+		Vector3Int clampedDir = direction.NormalizeTo3Int();
+		Vector3Int origin = ServerPosition;
+		Vector3Int roundedTarget = origin + clampedDir;
+
+		if (!ignorePassable && !MatrixManager.IsPassableAt(origin, roundedTarget, true, includingPlayers: !followMode))
+		{
+			return false;
+		}
+
+		if (isNewtonian && !MatrixManager.IsSlipperyOrNoGravityAt(roundedTarget))
+		{
+			return false;
+		}
+
+		if (!followMode && MatrixManager.IsEmptyAt(roundedTarget, true))
+		{
+			serverState.WorldImpulse = (Vector3)clampedDir;
+		}
+		else
+		{
+			serverState.WorldImpulse = Vector2.zero;
+		}
+
+		if (!float.IsNaN(speed) && speed > 0)
+		{
+			serverState.Speed = speed;
+		}
+		else
+		{
+			serverState.Speed = PushPull.DEFAULT_PUSH_SPEED;
+		}
+
+		if (followMode)
+		{
+			serverState.IsFollowUpdate = true;
+			SetPosition(roundedTarget);
+			serverState.IsFollowUpdate = false;
+		}
+		else
+		{
+			SetPosition(roundedTarget);
+		}
+
+		return true;
+	}
+
+	public bool PredictivePush(Vector2Int target, float speed = Single.NaN, bool followMode = false)
+	{
+		Poke();
+		Vector3Int target3int = target.To3Int();
+
+		Vector3Int currentPos = ClientPosition;
+
+		if (!followMode && !MatrixManager.IsPassableAt(target3int, target3int, isServer : false))
+		{
+			return false;
+		}
+
+		if (!followMode && MatrixManager.IsEmptyAt(target3int, false))
+		{
+			predictedState.WorldImpulse = target - currentPos.To2Int();
+		}
+		else
+		{
+			predictedState.WorldImpulse = Vector2.zero;
+		}
+
+		if (!float.IsNaN(speed) && speed > 0)
+		{
+			predictedState.Speed = speed;
+		}
+		else
+		{
+			predictedState.Speed = PushPull.DEFAULT_PUSH_SPEED;
+		}
+
+		predictedState.MatrixId = MatrixManager.AtPoint(target3int, false).Id;
+		predictedState.WorldPosition = target3int;
+
+		//		Lerp to compensate one frame delay
+		Lerp();
+
+		return true;
+	}
+
+	/// <summary>
+	/// Stop floating, kill impulse
+	/// </summary>
+	public void Stop()
+	{
+		Stop(true);
+	}
+
+	private void Stop( bool notify )
+	{
+		Logger.LogTraceFormat(STOPPED_FLOATING, Category.Transform, gameObject.name);
+        if (IsTileSnap)
+        {
+        	serverState.Position = Vector3Int.RoundToInt(serverState.Position);
+        }
+        else
+        {
+        	serverState.Speed = 0;
+        }
+        serverState.WorldImpulse = Vector2.zero;
+        serverState.SpinRotation = transform.localRotation.eulerAngles.z;
+        serverState.SpinFactor = 0;
+
+        if ( IsBeingThrownServer )
+        {
+			OnThrowEnd.Invoke(serverState.ActiveThrow);
+        }
+        serverState.ActiveThrow = ThrowInfo.NoThrow;
+        if ( notify )
+        {
+			NotifyPlayers();
+        }
+        registerTile.UpdatePositionServer();
+	}
+
+	public void OnClientStartFollowing(){}
+
+	public void OnClientStopFollowing(){}
```

Comment:
Codacy found an issue: [Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)